lib/ansible/modules/cloud/vmware/vmware_guest_disk.py

```diff
@@ -296,6 +376,25 @@ def reconfigure_vm(self, config_spec, device_type):
 
         return changed, results
 
+    def get_ioandshares_diskconfig(self, disk_spec, disk):
+        if 'iolimit' in disk:
+            io_disk_spec = vim.StorageResourceManager.IOAllocationInfo()
+            io_disk_spec.limit = disk['iolimit']['limit']
+            if 'shares' in disk['iolimit']:
+                shares_spec = vim.SharesInfo()
+                shares_spec.level = disk['iolimit']['shares']['level']
+                if shares_spec.level == 'custom':
+                    shares_spec.shares = disk['iolimit']['shares']['level_value']
+                io_disk_spec.shares = shares_spec
+            disk_spec.device.storageIOAllocation = io_disk_spec
+        if 'shares' in disk:
+            shares_spec = vim.SharesInfo()
```

Comment:
`SharesInfo` is deprecated since 4.1, as per document - https://vdc-repo.vmware.com/vmwb-repository/dcr-public/9e4aab9b-dd94-4213-8677-69e883bc8b6f/3aa4a23c-0253-452a-b750-3fe401fe74f9/doc/vim.vm.device.VirtualDisk.html#field_detail

Reply:
@ngp-star This is a deprecated feature, right?

Reply:
@Akasurde  
yes, and the link what you mentioned, in that it says shares is deprecated, instead of that use shares disk IO allocation spec type ,so i used shares of disk IO allocation spec.